projects/plugins/jetpack/modules/carousel/jetpack-carousel.js

```diff
@@ -1624,19 +1383,64 @@
 				settings.startIndex = 0; // -1 returned if can't find index, so start from beginning
 			}
 
-			domUtil.emitEvent( carousel.container, 'jp_carousel.beforeOpen' );
+			domUtil.emitEvent( carousel.overlay, 'jp_carousel.beforeOpen' );
+			carousel.gallery.innerHTML = '';
 
-			domUtil.fadeIn( carousel.container, function () {
-				domUtil.emitEvent( carousel.container, 'jp_carousel.afterOpen' );
-			} );
+			// Need to set the overlay manually to block or swiper does't initialise properly.
+			carousel.overlay.style.opacity = 1;
+			carousel.overlay.style.display = 'block';
 
-			carousel.gallery.innerHTML = '';
 			initCarouselSlides( gallery.querySelectorAll( settings.imgSelector ), settings.startIndex );
-			selectSlideAtIndex( settings.startIndex );
+
+			swiper = new window.Swiper( '.swiper-container', {
+				centeredSlides: true,
+				zoom: true,
+				loop: carousel.slides.length > 1 ? true : false,
+				pagination: {
+					el: '.swiper-pagination',
+					clickable: true,
+				},
+				navigation: {
+					nextEl: '.swiper-button-next',
+					prevEl: '.swiper-button-prev',
+				},
+				initialSlide: settings.startIndex,
+				on: {
+					init: function () {
+						selectSlideAtIndex( settings.startIndex );
+					},
+				},
+			} );
+
+			swiper.on( 'slideChange', function () {
+				var index;
+				// Swiper indexes slides from 1, plus when looping to left last slide ends up
+				// as 0 and looping to right first slide as total slides + 1. These are adjusted
+				// here to match index of carousel.slides.
+				if ( swiper.activeIndex === 0 ) {
+					index = carousel.slides.length - 1;
+				} else if ( swiper.activeIndex === carousel.slides.length + 1 ) {
+					index = 0;
+				} else {
+					index = swiper.activeIndex - 1;
+				}
+				selectSlideAtIndex( index );
+			} );
+
+			domUtil.fadeIn( carousel.overlay, function () {
+				domUtil.emitEvent( carousel.overlay, 'jp_carousel.afterOpen' );
+			} );
 		}
 
 		// Register the event listener for starting the gallery
 		document.body.addEventListener( 'click', function ( e ) {
+			var isIE11 = window.navigator.userAgent.match( /Trident\/7\./ );
+			// IE11 support is being dropped in August 2021. The new swiper.js libray is not IE11 compat
```

Comment:
@jeherve : Is this compatible with Jetpack's policies for a point release?

Reply:
Nitpick: s/libray/library/

Reply:
We're making a bit of an exception with IE11 right now, so that should be fine, yes.